free-2.md

```diff
@@ -1,44 +1,57 @@
 
+A DSL seems like a good choice because wine pairing is a fairly narrow domain, and users could use the language in order to educate themselves without having to go through too much other information. A language devoted strictly to this topic could be really helpful in assisting users expand their tastes for wine and food.
 
 ### Why you?
 _What excites you about this idea? How did you come up with it?_
 
+My whole family are huge wine snobs, yet I find that I am often lacking in the taste refinement that they have. I feel like I would totally use an app like this to improve my own experiences with food and wine, and I feel like even experienced wine drinkers could learn some new things about the way they experience wine.
 
 ### Domain
 _Describe the project's domain in five words._
 
+Pairing foods with appropriate wines
 
 ### Interface (syntax)
 _How might the user interact with the language? What does programming look 
 like? Why is this the right way to interact with the problem domain?_ 
 
+I think there are two ways one could go about this. One option is that the user inputs specific foods or recipes - spaghetti marinara, grilled fish with soy sauce, etc - and then the program would provide either wine categories or specific wines that would work well with that food/recipe. This could be a tuples of wines/wine categories, and the amount between 0 and 1 that that wine is a good match. As such, programming in that language would look something like searching through a list of foods and then calling a findPairings on that food. This would be nice because it would be easiest for the user to get exactly the desired result.
+
+Another option (probably the easier to program) would be to have the user input flavors or ingredients that are dominant in the food they are eating, and then have those inputs evaluated in order to produce the best matching wines. This seems like what would be happening under the surface anyway in the first case, and would require more work from the user. This would possibly be more versatile, however, if the user wants to pair with a food that isn't well known or perhaps doesn't even have a well known name.
 
 ### Operation (semantics)
 _What might happen when a program runs? How does a program interact with the
 user? What kinds of errors might occur, and how might they be communicated to
 the user?_
 
+Once the program receives the inputs from the user, it would then aggregate the flavors/recipes and use the dominant tastes/textures/whatever else in those foods in order to match them up with the best wine. I imagine it would create some sort of coefficient for each wine or wine category, which would describe how good of a fit that wine is. The highest scoring wines would then be returned to the user.
+
+Errors would occur if foods or ingredients are not recognized. I imagine the user would be prompted to add those foods to the DB, or otherwise to try something else.
```

Comment:
This really feels like an App in concept. I feel like striving for a GUI interface would greatly increase the usability of this DSL.